theano/gof/vm.py

```diff
@@ -332,7 +332,8 @@ def recursively_evaluate(var):
 
     def __init__(self, nodes, thunks, pre_call_clear,
                  storage_map, compute_map, fgraph, allow_gc,
-                 dependencies=None, callback=None, callback_input=None):
+                 dependencies=None, callback=None, callback_input=None,
+                 n_updates=0):
```

Comment:
I don't like that n_updates is 0 by default. Can you make it None by default and if the user use output_subset, then assert it isn't None? This would prevent some error hard to diagnose by the user.